packages/epo-widget-lib/src/atomic/Blinker/Blinker.tsx

```diff
@@ -1,28 +1,49 @@
-import { FunctionComponent, useState } from "react";
+import { FunctionComponent, useEffect, useState } from "react";
+import { tokens } from "@rubin-epo/epo-react-lib";
 import useInterval from "@/hooks/useInterval";
+import useResizeObserver from "use-resize-observer";
 import * as Styled from "./styles";
 import { Image } from "@rubin-epo/epo-react-lib/dist/types/image";
 import { getClampedArrayIndex } from "@/lib/utils";
 
 export interface BlinkerProps {
   images: Image[];
-  activeIndex: number;
+  activeIndex?: number;
   autoplay?: boolean;
   loop?: boolean;
   interval?: number;
-  blinkCallback: (activeIndex: number) => void;
+  blinkCallback?: (activeIndex: number) => void;
+  loadedCallback?: () => void;
+  className?: string;
 }
 
 const Blinker: FunctionComponent<BlinkerProps> = ({
   images = [],
-  activeIndex = 0,
+  activeIndex: presetIndex = 0,
   autoplay = true,
   loop = true,
   interval = 200,
   blinkCallback,
+  loadedCallback,
+  className,
 }) => {
+  const { ref, width = 1 } = useResizeObserver<HTMLDivElement>();
   const [playing, setPlaying] = useState(autoplay);
+  const [activeIndex, setActiveIndex] = useState(presetIndex);
+  const [loaded, setLoaded] = useState(false);
+  const { BREAK_MOBILE } = tokens;
   const canBlink = images.length > 1;
+  const isCondensed = width < parseInt(BREAK_MOBILE);
+
+  useEffect(() => {
+    blinkCallback && blinkCallback(activeIndex);
+  }, [activeIndex]);
+
+  useEffect(() => {
+    if (loaded) {
+      loadedCallback && loadedCallback();
```

Comment:
added this callback for when the blinker is loaded so that until then the elapsed time and click callbacks are not made in the source selector